third_party/bigframes_vendored/pandas/core/frame.py

```diff
@@ -2240,6 +2263,45 @@ def join(self, other, *, on: Optional[str] = None, how: str) -> DataFrame:
 
         Join columns with `other` DataFrame on index
 
+        **Examples:**
+
+            >>> import bigframes.pandas as bpd
+            >>> bpd.options.display.progress_bar = None
+
+        Join two DataFrames by specifying how to handle the operation:
+
+            >>> df1 = bpd.DataFrame({'col1': ['foo', 'bar'], 'col2': [1, 2]})
+            >>> df1
+               col1  col2
+            0   foo     1
+            1   bar     2
+            <BLANKLINE>
+            [2 rows x 2 columns]
+
+            >>> df2 = bpd.DataFrame({'col3': ['foo', 'baz'], 'col4': [3, 4]})
+            >>> df2
+               col3  col4
+            0   foo     3
+            1   baz     4
+            <BLANKLINE>
+            [2 rows x 2 columns]
+
+            >>> df1.join(df2, how="outer")
```

Comment:
This output will remain the same for `how="left"` as well. It would be nice to have asymmetry in the two input data frames and then showcase various styles of joins. E.g.

```python
>>> df1 = pd.DataFrame({'col1': ['foo', 'bar'], 'col2': [1, 2]}, index=[0, 1])
>>> df2 = pd.DataFrame({'col3': ['foo', 'baz'], 'col4': [3, 4]}, index=[1, 2])
>>> df1
  col1  col2
0  foo     1
1  bar     2
>>> df2
  col3  col4
1  foo     3
2  baz     4
>>> df1.join(df2)
  col1  col2 col3  col4
0  foo     1  NaN   NaN
1  bar     2  foo   3.0
>>> df1.join(df2, how="left")
  col1  col2 col3  col4
0  foo     1  NaN   NaN
1  bar     2  foo   3.0
>>> df1.join(df2, how="right")
  col1  col2 col3  col4
1  bar   2.0  foo     3
2  NaN   NaN  baz     4
>>> df1.join(df2, how="outer")
  col1  col2 col3  col4
0  foo   1.0  NaN   NaN
1  bar   2.0  foo   3.0
2  NaN   NaN  baz   4.0
>>> df1.join(df2, how="inner")
  col1  col2 col3  col4
1  bar     2  foo     3
>>> df1.join(df2, how="cross")
  col1  col2 col3  col4
0  foo     1  foo     3
1  foo     1  baz     4
2  bar     2  foo     3
3  bar     2  baz     4
>>>
```

Reply:
Done.